libcontainer/cgroups/fs/memory.go

```diff
@@ -19,14 +19,19 @@ type MemoryGroup struct {
 
 func (s *MemoryGroup) Apply(d *data) (err error) {
 	path, err := d.path("memory")
-	if err != nil {
-		if cgroups.IsNotFound(err) {
-			return nil
-		}
+	if err != nil && !cgroups.IsNotFound(err) {
 		return err
 	}
-	if err := os.MkdirAll(path, 0755); err != nil {
-		return err
+	if memoryAssigned(d.c) {
+		if path != "" {
+			if err := os.MkdirAll(path, 0755); err != nil {
```

Comment:
@hqhq 
if path not equal to empty( which means path exists), code is creating mkdir again with "path"
is that intentional ?